src/hotspot/share/cds/archiveHeapLoader.cpp

```diff
@@ -61,6 +61,10 @@ ptrdiff_t ArchiveHeapLoader::_mapped_heap_delta = 0;
 
 // Every mapped region is offset by _mapped_heap_delta from its requested address.
 // See FileMapInfo::heap_region_requested_address().
+// avoid adding to CompressedOops::base() == nullptr
```

Comment:
I don't understand what the comment is trying to say.

This is surely a bug in ubsan as I should be able to take any address, including zero, and  add an offset to get another address!

Reply:
Hi David, it is no ubsan-bug, see also here for more details about this warning  https://reviews.llvm.org/D67122 .